src/launcher/index.ts

```diff
@@ -109,7 +108,10 @@ async function main() {
       }
     })
 
-    ipcMain.on('flowr-desktop-config', (event: IpcMainEvent, desktopConfig?: { userPreferences: Partial<IFlowrStore>, player: Partial<IPlayerStore> }) => {
+    ipcMain.on('flowr-desktop-config', async (event: IpcMainEvent, desktopConfig?: IFlowrDesktopConfig) => {
+      if (desktopConfig?.userPreferences?.clearAppDataOnStart) {
```

Comment:
I would wait after the `initStore` function has been called. This way we perform the check after everything has been merged and we are certain to have a properly setup store.
```javascript
if (flowrWindow.store.get('userPreferences').clearAppDataOnStart)
```